primitives/chain-params/src/lib.rs

```diff
@@ -35,3 +35,40 @@ pub struct SidechainParams {
 	#[cfg_attr(feature = "std", arg(long))]
 	pub governance_authority: MainchainAddressHash,
 }
+
+pub fn default_chain_id() -> u16 {
+	1
+}
+pub fn default_numerator() -> u64 {
+	2
+}
+pub fn default_denominator() -> u64 {
+	3
+}
+
+#[cfg(feature = "std")]
+impl SidechainParams {
+	pub fn read_from_env_with_defaults() -> Result<Self, envy::Error> {
+		/// This structure is needed to read sidechain params from the environment variables because the main
+		/// type uses `rename_all = "camelCase"` serde option
+		#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+		struct SidechainParamsEnvConfiguration {
+			#[serde(default = "default_chain_id")]
+			pub chain_id: u16,
+			pub genesis_committee_utxo: UtxoId,
+			#[serde(default = "default_numerator")]
+			pub threshold_numerator: u64,
+			#[serde(default = "default_denominator")]
+			pub threshold_denominator: u64,
+			pub governance_authority: MainchainAddressHash,
+		}
+		let raw = envy::from_env::<SidechainParamsEnvConfiguration>()?;
```

Comment:
Does it log the fact that the default value was used? It might be helpful when debugging configuration related issues

Reply:
It doesn't, unfortunately